core/iwasm/interpreter/wasm_loader.c

```diff
@@ -15460,8 +15460,15 @@ wasm_loader_prepare_bytecode(WASMModule *module, WASMFunction *func,
 #endif
                 if (opcode1 != WASM_OP_ATOMIC_FENCE) {
                     CHECK_MEMORY();
-                    read_leb_uint32(p, p_end, align);      /* align */
-                    read_leb_uint32(p, p_end, mem_offset); /* offset */
+                    read_leb_uint32(p, p_end, align); /* align */
+#if WASM_ENABLE_MEMORY64 != 0
+                    is_memory64 = module->memories[0].flags & MEMORY64_FLAG;
```

Comment:
It may be import memory, had better check whether `module->import_memory_count > 0` or not